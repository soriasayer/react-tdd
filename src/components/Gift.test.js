import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Gift from './Gift';

configure({ adapter: new Adapter(), disableLifecycleMethods: true });

describe('Gift', () => {
  const gift = shallow(<Gift/>)

  it('render properly', () => {
    expect(gift).toMatchSnapshot()
  })

  it('initialize a person and present in `state`', () => {
    expect(gift.state()).toEqual({person: '', present: ''})
  })

  describe('when typing into the person `input`', () => {
    beforeEach(() => {
      gift.find('.input-person').simulate('change', {target: {value: 'Mom'}})
    })

    it('update the person in `state`', () => {
      expect(gift().this.state()).toEqual('Mom')
    })
  })
})
