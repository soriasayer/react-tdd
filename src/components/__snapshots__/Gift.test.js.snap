// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift render properly 1`] = `ShallowWrapper {}`;
